xarray/backends/api.py

```diff
@@ -487,10 +489,40 @@ def maybe_decode_store(store, lock=False):
             )
             name_prefix = "open_dataset-%s" % token
             ds2 = ds.chunk(chunks, name_prefix=name_prefix, token=token)
-            ds2._file_obj = ds._file_obj
+
+        elif engine == "zarr":
+            # adapted from Dataset.Chunk() and taken from open_zarr
+            if not (isinstance(chunks, (int, dict)) or chunks is None):
+                if chunks != "auto":
+                    raise ValueError(
+                        "chunks must be an int, dict, 'auto', or None. "
+                        "Instead found %s. " % chunks
+                    )
+
+            if chunks == "auto":
+                try:
+                    import dask.array  # noqa
+                except ImportError:
+                    chunks = None
+
+            # auto chunking needs to be here and not in ZarrStore because
+            # the variable chunks does not survive decode_cf
+            # return trivial case
+            if not chunks:  # e.g. chunks is 0, None or {}
```

Comment:
Cam we make this `if chunks is None` instead?

I know this is a discrepancy from how `open_zarr()` works today, but currently `open_dataset(..., chunks={})` is a way to open a dataset with dask chunks equal to the full size of any arrays.

I doubt the (different) behavior of `open_zarr` in this case was intentional....

Reply:
I'll try it locally and see if the tests break, felt like there was a reason it had to be `not chunks` but can't remember the context now.

Reply:
Thanks! Not a big deal if we have to push off this clean-up.

Reply:
Yep, 2 tests would break with a `ZeroDivisionError` if we switch to `if chunks is None`. Specifically:

- TestZarrDictStore.test_manual_chunk
- TestZarrDirectoryStore.test_manual_chunk

Related to my comment hidden in the mess above at https://github.com/pydata/xarray/pull/4187#issuecomment-652104356 :smile:

Reply:
Well, I would expect `test_manual_chunk` to fail here: it is explicitly verifying that `chunks=0` and `chunks={}` result in-memory numpy arrays. Does it work if you remove those cases, e.g., by setting `NO_CHUNKS = (None,)` ?

Reply:
Yes, setting `NO_CHUNKS = (None,)` works with `if chunks is None`. I'll make the change?

Reply:
If it helps by the way, `test_manual_chunk` with `NO_CHUNKS = (None, 0, {})` was added in #2530.

Reply:
Yes, I think we can go ahead and change that. It doesn't look like that was carefully evaluated in #2530.